### DISCUSSION TITLE
Langchain-Chatchat v0.2.0

### DISCUSSION THREAD

**Discussion by imClumsyPanda** [chatchat-space/Langchain-Chatchat#1062] [Announcements]
## 重要提示
因 `v0.2.0` 版本与前序版本相比调整较大，使用 `v0.2.0` 版本时建议先新建环境，重新安装依赖包后，按照readme中开发部署操作顺序进行配置。

自版本 `v0.2.0` 起，项目正式更名为 `Langchain-Chatchat`，项目将基于 Langchain 和包含 ChatGLM-6B 在内的开源大语言模型提供持续优化的知识库问答应用。

## 变更信息
* 使用 FastChat 提供开源 LLM 模型的 API，以 OpenAI API 接口形式接入，提升 LLM 模型加载效果，并支持近期新出的baichuan-inc/baichuan-7B、internlm/internlm-chat-7b、Qwen/Qwen-7B-Chat等模型；

* 使用 langchain 中已有 Chain 的实现，便于后续接入不同类型 Chain，并将对 Agent 接入开展测试；

* 使用 FastAPI 提供 API 服务，全部接口可在 FastAPI 自动生成的 docs 中开展测试，且所有对话接口支持通过参数设置流式或非流式输出；

* 使用 Streamlit 提供 WebUI 服务，可选是否基于 API 服务启动 WebUI，后续可支持不同形式输出内容的显示；

* 项目中默认 LLM 模型改为 THUDM/chatglm2-6b，默认 Embedding 模型改为 moka-ai/m3e-base，文件加载方式与文段划分方式也有调整，后续将重新实现上下文扩充，并增加可选设置；

* 项目中扩充了对不同类型向量库的支持，除支持 FAISS 向量库外，还提供 Milvus, PGVector 向量库的接入；

* 项目中搜索引擎对话，除 Bing 搜索外，增加 DuckDuckGo 搜索选项，DuckDuckGo 搜索无需配置 API Key，在可访问国外服务环境下可直接使用。

## 主要贡献者
@imClumsyPanda 
@liunux4odoo 
@zqt996 
@hzg0601 
@glide-the 
@inksong 

<hr /><em>This discussion was created from the release <a href='https://github.com/chatchat-space/Langchain-Chatchat/releases/tag/v0.2.0'>Langchain-Chatchat v0.2.0</a>.</em>

**Comment by guoqingkong** [comment]
0.2版本按照readme执行1、2、3、4、5，在根目录运行 llm_api.py 后无法启动，找不到'configs.model_config'包

**Reply by imClumsyPanda** [reply]
请参考 https://github.com/chatchat-space/Langchain-Chatchat/issues/1084#issuecomment-1677179824 ，步骤3中有说明如何把model_config.py.example复制后重命名为model_config.py并进行相应修改

**Reply by guoqingkong** [reply]
有这个文件，还出现这个情况的：

(langchain) kong@kong-Super-Server:~/dev/langchain-ChatGLM$ tree configs
configs
├── __init__.py
├── model_config.py
├── model_config.py.example
└── __pycache__
    ├── __init__.cpython-310.pyc
    └── model_config.cpython-310.pyc

1 directory, 5 files
(langchain) kong@kong-Super-Server:~/dev/langchain-ChatGLM$ python server/llm_api.py
Traceback (most recent call last):
  File "/home/kong/dev/langchain-ChatGLM/server/llm_api.py", line 6, in <module>
    from configs.model_config import llm_model_dict, LLM_MODEL, LLM_DEVICE, LOG_PATH, logger
ModuleNotFoundError: No module named 'configs.model_config'
